tests/integration/s2n_dynamic_record_size_test.py

```diff
@@ -14,7 +14,19 @@
 #
 
 """
-Dynamic record size tests using s2nc against Openssl s_server
+This is an integration test for the use of dynamic record sizes in TLS connections.
+
+The function s2n_connection_set_dynamic_record_threshold can be used to dynamically change the size of TCP packets to
+optimize for both latency and throughput. This is done by first setting a threshold, until this threshold has been
+met the connection uses small TLS records that fit into a single TCP segment (mss). This optimizes the connection
+for low latency. Once the number of bytes transferred in the connection has met this threshold, the size of the
+records can exceed this maximum bound - in order to optimize throughput.
+
+This test sets up an s2nc connection against OpenSSL s_server to transfer a test file using each of the cipher
+suites supported. In each connection, we test:
+ - That all segment sizes before the threshold is met are less than the mss (usually mss = 1460B, but if the mss cannot
+ be obtained a default of 65496B is used).
+ - That at least one segment size after the threshold has been met is greater than 1500B.
```

Comment:
I might be a little confused here but if you're using the default of 65,496B wont all segments be bigger than 1,500B? Should the second point say "one (all?) segments after the threshold is greater than the mss"?

Where does 1,500B come from?

Reply:
It looks like due to the volatile nature of recording our TCP dump, that occasionally we miss the first/start message and therefore cannot recover the mss value set. In this case, the packets will still abide by the value agreed, we just don't know what it is. Therefore the test defaults to 65,496B and in this case gets a "free" pass - this is how the test has always ran.

Similarly with the 1500B, this is unchanged from the original test written by @fatrat1117. It looks like 1500B is the MTU (Maximum Transmission Unit) and thus if we are exceeding that we must be in high throughput. Ideally, we would like to set this as the mss, but I assume in the case where we miss the first message (and thus the agreed mss) we would auto fail the test.

Reply:
Thinking about it, missing the first few messages of the TCP dump could be even more problematic... if we're keeping a cumulative total of bytes transferred, but can't see a complete record of the TCP dump, then our total is inaccurate too. 

Most of the time this shouldn't have too much of an impact, as the first ~7 segments are usually 0 or ~200B (and they are usually the ones that don't get picked up in the dump). I imagine this is also partly what was causing the flakyness of the test to begin with, as back when the segment sizes were hard coded - missing a few segments would be even more destructive.

Reply:
Regardless of what we do with the default value can you make all the 1500's use a new `MTU_bytes` constant.

Reply:
done